java/daikon/dcomp/DCInstrument.java

```diff
@@ -1048,7 +1049,9 @@ public JavaClass instrument() {
     // Chicory runtime for this information.
     if (track_class) {
       // System.out.printf ("adding class %s to all class list%n", class_info);
-      daikon.chicory.Runtime.all_classes.add (class_info);
+      synchronized(daikon.chicory.Runtime.all_classes) {
```

Comment:
Here and below, add a space after `synchronized` so it doesn't look like a function call.